src/payment/strategies/checkoutcom/checkoutcom-google-pay-payment-strategy.spec.ts

```diff
@@ -0,0 +1,249 @@
+            googlePayPaymentProcessor
+        );
+
+        container = document.createElement('div');
+        walletButton = document.createElement('a');
+        container.setAttribute('id', 'login');
+        walletButton.setAttribute('id', 'mockButton');
+        document.body.appendChild(container);
+        document.body.appendChild(walletButton);
+
+        jest.spyOn(walletButton, 'removeEventListener');
+        jest.spyOn(requestSender, 'post').mockReturnValue(Promise.resolve());
+        jest.spyOn(checkoutActionCreator, 'loadCurrentCheckout').mockReturnValue(Promise.resolve());
+        jest.spyOn(paymentMethodActionCreator, 'loadPaymentMethod').mockReturnValue(Promise.resolve(store.getState()));
+        jest.spyOn(googlePayPaymentProcessor, 'initialize').mockReturnValue(Promise.resolve());
+        jest.spyOn(strategy, 'requestCheckoutcomTokenize').mockReturnValue(Promise.resolve(checkoucomTokenResponse));
+    });
+
+    afterEach(() => {
+        document.body.removeChild(container);
+        document.body.removeChild(walletButton);
+    });
+
+    it('creates an instance of GooglePayPaymentStrategy', () => {
+        expect(strategy).toBeInstanceOf(CheckoutcomGooglePayPaymentStrategy);
+    });
+
+    describe('#execute', () => {
+        let googlePayOptions: PaymentInitializeOptions;
+
+        beforeEach(() => {
+            jest.spyOn(walletButton, 'addEventListener');
+            jest.spyOn(store, 'dispatch').mockReturnValue(Promise.resolve());
+            googlePayOptions = {
+                methodId: 'googlepaycheckoutcom',
+                googlepaycheckoutcom: {
+                    walletButton: 'mockButton',
+                    onError: () => {},
+                    onPaymentSelect: () => {},
+                },
+            };
+        });
+
+        it('creates the order and submit payment', async () => {
+
+            jest.spyOn(orderActionCreator, 'submitOrder').mockReturnValue(Promise.resolve());
+            jest.spyOn(paymentActionCreator, 'submitPayment').mockReturnValue(Promise.resolve());
+            jest.spyOn(store.getState().paymentMethods, 'getPaymentMethod').mockReturnValue(googlePaymentMethodData);
+            jest.spyOn(store.getState().paymentMethods, 'getPaymentMethodOrThrow').mockReturnValue(googlePaymentMethodData);
+
+            await strategy.initialize(googlePayOptions);
+            await strategy.execute(getGoogleOrderRequestBody());
+
+            expect(orderActionCreator.submitOrder).toHaveBeenCalled();
+            expect(paymentActionCreator.submitPayment).toHaveBeenCalled();
+        });
+
+        it('throws and error when GooglePaymentOptions are not available', () => {
+            expect(() => strategy.execute(getGoogleOrderRequestBody())).toThrowError('Unable to initialize payment because "options.googlepaycheckoutcom" argument is not provided.');
+        });
+
+        it('gets again the payment information and submit payment', async () => {
```

Comment:
this test has the same assertions than `creates the order and submit payment`.
What's exactly different in this test? If we expect a same behaviour with a different pre-condition, might be worth saying something like
`it('creates the order and submit payment when X')`

Reply:
I fixed this and, subsequent tests, that where missing an expect, and paymentMethodData that should have been incomplete, for the test to call displayWallet